lnrpc-ideal/routerrpc/router.proto

```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+import "rpc.proto";
+
+package routerrpc;
+
+option go_package = "github.com/lightningnetwork/lnd/lnrpc/routerrpc";
+
+message TrackPaymentRequest {
+    /// The hash of the payment to look up.
+    bytes payment_hash = 1;
+}
+
+service Router {
+    /**
+    TrackPayment returns an update stream for the payment identified by the
+    payment hash.
+    */
+    rpc TrackPayment (TrackPaymentRequest) returns (stream lnrpc.Payment);
```

Comment:
Wondering whether there is a use case for `TrackAllPayments`. And if not, is the presence of `Subscribe(All)Invoices` legitimate?

Reply:
One difference is that payments are always initiated by the user, while invoices can be created automatically with keysend.

Reply:
Also every `SubscribeSingleInvoice` call creates a goroutine, so monitoring a huge amount of invoices simultaneously may have a performance impact

Reply:
I think the use case would be services which want to restore their subscriptions after restart? Rather than going through every unresolved invoice/payment and individually re-subscribing.

For invoices, it's nice that you can subscribe from your last known settle index. That way you don't miss out on anything. 
For payments, a track all payments would need a similar mechanism so that you can resume from your last restart. Don't want to just subscribe to all unresolved because you miss things that completed while your service was offline.